admin/src/components/DocumentPreviewModal.tsx

```
@@ -528,14 +636,28 @@ export default function DocumentPreviewModal({
+            title={fileName}
+          />
+        );
+      }
+
+      const ext = getFileExtension(fileType, fileUrl, fileName);
+      if (['DOCX', 'XLSX'].includes(ext) && officeHtml !== null) {
+        return (
+          <div className="w-full h-full overflow-auto bg-white p-6">
+            <div
+              className="max-w-none"
+              dangerouslySetInnerHTML={{ __html: officeHtml }}
```

Biome (2.3.14)

[error] Avoid passing content using the dangerouslySetInnerHTML prop.

Setting content using code can expose users to cross-site scripting (XSS) attacks

(lint/security/noDangerouslySetInnerHtml)